core/src/main/java/org/apache/druid/data/input/impl/HttpEntity.java

```diff
@@ -80,23 +80,30 @@ protected String getPath()
     return t -> t instanceof IOException;
   }
 
-  public static InputStream openInputStream(URI object, String userName, PasswordProvider passwordProvider, long offset)
-      throws IOException
-  {
-    final URLConnection urlConnection = object.toURL().openConnection();
+  private static void addAuthHeader(URLConnection urlConnection, String userName, PasswordProvider passwordProvider){
     if (!Strings.isNullOrEmpty(userName) && passwordProvider != null) {
       String userPass = userName + ":" + passwordProvider.getPassword();
       String basicAuthString = "Basic " + Base64.getEncoder().encodeToString(StringUtils.toUtf8(userPass));
       urlConnection.setRequestProperty("Authorization", basicAuthString);
     }
+  }
+
+  public static InputStream openInputStream(URI object, String userName, PasswordProvider passwordProvider, long offset)
+      throws IOException
+  {
+    final URLConnection urlConnection = object.toURL().openConnection();
+    addAuthHeader(urlConnection, userName, passwordProvider);
     final String acceptRanges = urlConnection.getHeaderField(HttpHeaders.ACCEPT_RANGES);
     final boolean withRanges = "bytes".equalsIgnoreCase(acceptRanges);
     if (withRanges && offset > 0) {
       // Set header for range request.
       // Since we need to set only the start offset, the header is "bytes=<range-start>-".
       // See https://tools.ietf.org/html/rfc7233#section-2.1
-      urlConnection.addRequestProperty(HttpHeaders.RANGE, StringUtils.format("bytes=%d-", offset));
-      return urlConnection.getInputStream();
+      urlConnection.getInputStream().close();
```

Comment:
Doesn't this end up connecting to the destination server, and immediately closing that connection? Is that wanted? This is not "retrying" as the first connection is not fully valid, and we do no error checking. What do we mean by "retry" here?

Reply:
You are right, @paul-rogers , I don't think this is a retry case either.

The connection is being closed here because otherwise the `addRequestProperty(HttpHeaders.RANGE, ...)` would throw an exception as the connection is already open.

Reply:
@tejaswini-imply , please add a comment here mentioning why we need to close this connection.

Reply:
A couple of doubts : 
1. maybe we can use try-with-resources to manage the connection so that there's no leak incase of any exception
2. is it possible to determine the `ACCEPT-RANGES` value from a `HEAD` call? If we can do so, then one positive is that we can complete the requests we do to the server. but i'm not sure if that's possible

Reply:
There is another way that doesn't involve `ACCEPT-RANGES`. As per https://datatracker.ietf.org/doc/html/rfc7233#section-2.1, if a server is returning a partial response, then the server MUST generate a content-range header. The value of this header can be used to determine whether to skip offset or not.

Reply:
@kfaraz Previously, since new connection approach is deemed to be appropriate the old connection was closed before opening new one. Like Abhishek mentioned `Content-Range` header is more reliable than `Accept-Ranges` as per https://datatracker.ietf.org/doc/html/rfc7233#section-4, using this we need only single request to fetch content and decide whether it is partial or not.